src/qs_scf.F

```diff
@@ -541,6 +541,11 @@ SUBROUTINE scf_env_do_scf(scf_env, scf_control, qs_env, converged, should_stop)
 
       IF (.NOT. converged) CPWARN("SCF run NOT converged")
 
+      IF (.NOT. converged .AND. scf_control%stop_higher_iter_level) THEN
+        logger%iter_info%last_iter(logger%iter_info%n_rlevel-1) = .TRUE.
+        CPWARN("MD iteration also stops")
```

Comment:
this functionality does not seem to be `MD` exclusive and `scf_env_do_scf` is also called by `xas_tp_scf`

Reply:
I will check, but should not be effective there